numpy/lib/index_tricks.py

```diff
@@ -840,6 +840,26 @@ def fill_diagonal(a, val, wrap=False):
            [0, 4, 0, 0, 0],
            [0, 0, 4, 0, 0]])
 
+    Anti-diagonal example:
+    The anti-diagonal can be obtained by reversing the order of elements
+    using either `numpy.flipud` or `numpy.fliplr`.
+
+    >>> a = np.zeros((3,3))
+    >>> a
+    array([[0., 0., 0.],
+           [0., 0., 0.],
+           [0., 0., 0.]])
+    >>> np.fill_diagonal(np.fliplr(a), 2) # Horizontal flip
```

Comment:
These examples would be better if you filled with `[1,2,3]` so the direction is visible

Reply:
I did it that way initially, then I have sent the forced update because I thought it was clear enough that the second example was overriding with "9"  all the "2". Rolling back then..

Reply:
I think you misunderstood this comment. I was not asking for the original array to be filled with non-zeros